events/create_handler.go

```diff
@@ -0,0 +1,44 @@
+package events
+
+import (
+	log "github.com/Sirupsen/logrus"
+	"github.com/fsouza/go-dockerclient"
+	"github.com/rancherio/go-machine-service/locks"
+	rclient "github.com/rancherio/go-rancher/client"
+)
+
+type CreateHandler struct {
+	client   SimpleDockerClient
+	rancher  *rclient.RancherClient
+	hostUuid string
+}
+
+func (h *CreateHandler) Handle(event *docker.APIEvents) error {
+	// Note: event.ID == container's ID
+	lock := locks.Lock("create." + event.ID)
+	if lock == nil {
+		log.Warnf("Container locked. Can't run CreateHandler. ID: [%s]", event.ID)
+		return nil
+	}
+	defer lock.Unlock()
+
+	container, err := h.client.InspectContainer(event.ID)
+	if err != nil {
+		return err
+	}
+
+	containerEvent := &rclient.ContainerEvent{
+		ExternalStatus:    event.Status,
+		ExternalId:        event.ID,
+		ExternalFrom:      event.From,
+		ExternalTimestamp: int(event.Time),
```

Comment:
Time is in seconds.

Status is event name like `create` or `start`.
From is the image.

I actually thought it best to use the same names in our resource *because* of the weird names in the docker API. I wanted our resources to be just an simple, unopinionated copy of the docker event. The opinion comes in our handlers that act upon the event.